Contrast 18474 update contrast dot net sdk

rduran0 opened the pull request:
Updated SDK to handle most recent version of the API:

* Added response classes for all requests methods.
* Updated objects that already existed and most of their field names.
* Removed most deprecated fields.
* Added new urls to NgEndpoints. Deprecated previous Url class Endpoints.
* Deprecated profile request methods.
* Added filters for Server and Trace request methods to specify extra query params.
* Added new request methods to retrieve Trace events data, trace story and trace http request.
* Updated existing tests and created new ones for new methods.
* Updated Program.cs class with new flow. Replace CheckForTrace example with GetTraceByUUID.

## Notes

* Some old fields were left due to being displayed on web UI but not displayed on documentation.
* All unused old fields that weren't removed, were marked as deprecated.

rduran0 committed:
CONTRAST-18474

* Added new endpoint urls.
* Updated traces, applications and servers request methods.
* Updated Trace, Application and Server classes.
* Added response for Trace, Application and Server classes. Includes others resources used by them.

rduran0 committed:
CONTRAST-18474

Updated endpoint methods. Deprecated those methods that might not be useful anymore or require to be remade completely.

rduran0 committed:
CONTRAST-18474

* Added Server filter.
* Created unit tests for filters.

rduran0 committed:
CONTRAST-18474

* Updated example Program.
* Update TeamServerClientTest.
* Added new methods and models to request trace events.

rduran0 committed:
CONTRAST-18474

* Added new tests for trace events endpoints.
* Minor fixes to new methods and objects.

SteveFeldman commented:
@rduran0 seems like now is the perfect time to add a Travis job for this eh?

rduran0 committed:
CONTRAST-18474

Updated RuleSeverity names. Changed to uppercase to avoid Server Error.

gmoore commented on SampleContrastClient/Program.cs:
```diff
@@ -86,7 +86,7 @@ static void Main(string[] args)
 
                         //foreach (Trace trace in traces)
                         //{
-                        //    Console.WriteLine("Trace Exists:{0}", DoesTraceExistForUrl(client, appId, trace.Request.Uri));
```
Just curious: why are we updating commented code?

rduran0 replied:
I think it was meant to work as example hence why I updated.

gmoore commented on SampleContrastClient/Program.cs:
```diff
@@ -137,13 +137,12 @@ private static void DownloadAgentToDesktop(TeamServerClient client)
-        // Example usage of CheckForTrace method
-        private static bool DoesTraceExistForUrl( TeamServerClient client, string applicationId, string url )
+        // Example usage of GetTracesByUUID method
+        private static bool GetTraceByUUID( TeamServerClient client, string traceUuid, string organizationId )
```
If we're not returning the actual trace here, let's rename this method. Something `DoesTraceExist`

rduran0 replied:
You are right. I'll change it back.

gmoore reviewed (changes requested):
Just a few q's

rduran0 committed:
CONTRAST-18474

* Fixed wrong response for GetApplicationTraces and GetServerTraces.
* Added new tests for the mentioned methods.

rduran0 committed:
CONRAST-18474

Added appveyor.yml to enable AppVeyor CI.

rduran0 committed:
CONTRAST-18474

Appveyor.yml fix.

rduran0 commented:
@SteveFeldman @gmoore I added AppVeyor CI for building this project. I was planing to use Travis at first but then I read in the documents that [Travis builds C# projects with Mono or .NET Runtime in Linux or OS X and it might not be fully compatible with Windows runtimes](https://docs.travis-ci.com/user/languages/csharp/#Build-Environment).

It runs tests also by default. I omitted the deployment part for now since its completely unknown to me what process you follow to deploy it to Nuget.

rduran0 committed:
CONTRAST-18474

Updated method name.

rduran0 committed:
CONTRAST-18474

Updated most methods that returned lists. They now return the whole response object with the list.

rduran0 committed:
CONTRAT-18474

Updated sample program class.

gmoore commented:
Do you have a link to the project on AppVeyor?

rduran0 commented:
Sure, here is last run: https://ci.appveyor.com/project/rduran0/contrast-sdk-dotnet
